src/norm/mongodb.nim

```diff
@@ -0,0 +1,727 @@
+# From BSON to Nim Types:
+#
+# | BSON          | Type      | Notes
+# | ------------- | --------- | ---------------------
+# | Double        | float     | default float in Nim already 64-bit
+# | String (UTF8) | string    | string already UTF-8
+# | Oid           | Oid       | https://nim-lang.org/docs/oids.html
+# | Bool          | bool      | 
+# | TimeUTC       | timestamp | https://nim-lang.org/docs/times.html
+# | Null          | Option[T] | Only fields of Option[T] can be null
+# | Int32         | int       | ideally, int32, but it will convert to int, int32, or int64
+# | Int64         | int       | ideally, int64, bit it will convert to int, int32, or int64
+#
+# NOT YET SUPPORTED
+#
+# | Document      | (Object)  | field points to another object type. the other object need not be "dressed" with `db`
+# | Array         | seq[T]    | we ONLY support same-type arrays right now
+# Array (heterogeneous lists not supported yet)
+# DBPointer
+# Binary
+# Timestamp
+#
+# NOT SUPPORTED (ignored):
+#
+# Regexp
+# JSCode
+# JSCodeWithScope
+# MaximumKey
+# MinimumKey
+#
+# From Nim Object to BSON
+#
+# | Type      | BSON          | Notes
+# | --------- | ------------- |---------------------
+# | float     | Double        | default float in Nim already 64-bit
+# | string    | String (UTF8) | string already UTF-8
+# | Oid       | Oid           | https://nim-lang.org/docs/oids.html; all-zeroes count as a MISSING entry; not a null
+# | bool      | Bool          | 
+# | timestamp | TimeUTC       | https://nim-lang.org/docs/times.html; 1970-01-01 00:00:00 counts as MISSING; not null
+# | Option[T] | Null          | Only fields of Option[T] can be null
+# | int       | Int32         |
+#
+# NOT YET SUPPORTED (compile-time error generated):
+#
+# | float32   | Double        | 
+# | float64   | Double        | default float in Nim already 64-bit
+# | (Object)  | Document      | it is not possible to a "Missing" state. The closest analogue is Option[ObjectType]
+# | seq[T]    | Array         | to support null, make it seq[Option[T]]
+# | int32     | Int32         |
+# | int64     | Int64         |
+# Tuples
+# Tables
+#
+
+import
+  strutils,
+  sequtils,
+  macros,
+  typetraits,
+  logging,
+  options,
+  tables,
+  oids,
+  times
+
+import
+  nimongo/bson,
+  nimongo/mongo
+
+import
+  rowutils,
+  objutils,
+  pragmas
+
+
+export
+  strutils,
+  sequtils,
+  macros,
+  logging,
+  options,
+  tables,
+  oids,
+  times
+
+export
+  bson,
+  mongo
+
+export
+  rowutils,
+  objutils,
+  pragmas
+
+# compile-time globals
+var
+  normObjectNamesRegistry {.compileTime.}: seq[string] = @[]
+
+const NORM_UNIVERSAL_TYPE_LIST* = @[
+  "float",
+  "string",
+  "Oid",
+  "bool",
+  "Time",
+  "int",
+  "Option[float]",
+  "Option[string]",
+  "Option[Oid]",
+  "Option[bool]",
+  "Option[Time]",
+  "Option[int]",
+  "seq[float]",
+  "seq[string]",
+  "seq[Oid]",
+  "seq[bool]",
+  "seq[Time]",
+  "seq[int]"
+]
+
+proc getCollectionName*(objRepr: ObjRepr): string =

```

Comment:
I can't decide what the right solution is. On one hand, there are no tables in Mongo, there are collections. Thus, it's logical to have `collection` pragma. On the other hand, since this is part of the bigger package, we want things consistent and similarly named, e.g. `table` and `getTableName` even if it's not technically a table.

Reply:
I'll make an alias. That makes sense.